ArduCopter/events.cpp

```diff
@@ -113,10 +136,24 @@ void Copter::failsafe_gcs_check()
     if (should_disarm_on_failsafe()) {
         init_disarm_motors();
     } else {
-        if (control_mode == AUTO && g.failsafe_gcs == FS_GCS_ENABLED_CONTINUE_MISSION) {
-            // continue mission
-        } else if (g.failsafe_gcs != FS_GCS_DISABLED) {
-            set_mode_RTL_or_land_with_pause(MODE_REASON_GCS_FAILSAFE);
+        if (!AP_Notify::flags.autopilot_mode && (g2.fs_gcs_options & ParametersG2::FS_GCS_OPT_PILOT_FLIGHT_CONTINUE)) {
```

Comment:
If I understood this correctly `AP_Notify::flags` should never be used outside of that module.